### PR TITLE
[GEP-18] Add documentation for the shoot cluster CA certificate rotation

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area security documentation
/kind enhancement

**What this PR does / why we need it**:
This PR adds a new document related to the rotation of the shoot cluster CA certificates.
Essentially, there is now one file explaining the rotation of all the various shoot credentials.
Hence, on the way I harmonized our documents a bit, so I recommend using the separate commits for the review.

**Which issue(s) this PR fixes**:
Part of #3292

**Release note**:

```doc user
A [new document](https://github.com/gardener/gardener/tree/master/docs/usage/shoot_credentials_rotation.md#certificate-authorities) related to the rotation of the CA certificate rotation has been added.
```

### CODE REVIEW

File: docs/usage/shoot_credentials_rotation.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+Depending on the cloud provider, the required [data keys of the `Secret` differ](https://github.com/gardener/gardener/blob/master/example/70-secret-provider.yaml).
+Please consult the documentation of the respective provider extension documentation to get to know the concrete data keys (e.g., [this document for AWS](https://github.com/gardener/gardener-extension-provider-aws/blob/master/docs/usage-as-end-user.md#provider-secret-data)).
+
+**It is the responsibility of the end-user to regularly rotate those credentials.**
+The following steps are required to perform the rotation:
+
+1. Update the data in the `Secret` with new credentials.
+2. ⚠️ Wait until all `Shoot`s using the `Secret` are reconciled before you disable the old credentials in your cloud provider account! Otherwise, the `Shoot`s will no longer work as expected. Check out [this document](shoot_operations.md#immediate-reconciliation) to learn how to trigger a reconciliation of your `Shoot`s.
+3. After all `Shoot`s using the `Secret` were reconciled, you can go ahead and deactivate the old credentials in your provider account account.
```
Comment (acumino) [inline docs/usage/shoot_credentials_rotation.md:20]:
```suggestion
3. After all `Shoot`s using the `Secret` were reconciled, you can go ahead and deactivate the old credentials in your provider account.
```

File: docs/usage/shoot_credentials_rotation.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+> ⚠️ This does not invalidate the old client certificate. In order to do this, you should perform a rotation of the CAs (see section below).
+
+**It is the responsibility of the end-user to regularly rotate those credentials (or disable this `kubeconfig` entirely).**
+In order to rotate the `token` in this `kubeconfig`, annotate the `Shoot` with `gardener.cloud/operation=rotate-kubeconfig-credentials`.
+This operation is not allowed for `Shoot`s that are already marked for deletion.
+Please note that only the token (and basic auth password, if enabled) are exchanged.
+The CA certificate remains the same (see section below for information about the rotation).
+
+```bash
+kubectl -n garden-<project-name> annotate shoot <shoot-name> gardener.cloud/operation=rotate-kubeconfig-credentials
```
Comment (ialidzhikov) [inline docs/usage/shoot_credentials_rotation.md]: Nit: the project Namespace could be different from `garden-<project-name>` in some cases. If the Project `.spec.namespace` is omitted on creation, it should use a Namespace name like `garden-<project-name>-<random-chars>`.

We could instead use `<project-namespace>` or `<shoot-namespace>`.

```suggestion
kubectl -n <project-namespace> annotate shoot <shoot-name> gardener.cloud/operation=rotate-kubeconfig-credentials
```

**Review by ialidzhikov — COMMENTED** [review]
Nice docs!

/lgtm

File: docs/usage/shoot_credentials_rotation.md [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+In order to start the rotation (stage one), you have to annotate the shoot with the `rotate-ca-start` operation:
+
+```bash
+kubectl -n <shoot-namespace> annotate shoot <shoot-name> gardener.cloud/operation=rotate-ca-start
+```
+
+This will trigger a `Shoot` reconciliation and performs stage one.
+After it is completed, the `.status.credentials.rotation.certificateAuthorities.phase` is set to `Prepared`.
+
+Now you must update all API clients outside the cluster (such as the `kubeconfig`s on developer machines) to use the newly issued CA bundle in the `<shoot-name>.ca-cluster` `Secret`.
```
Comment (timebertt) [inline docs/usage/shoot_credentials_rotation.md:91]: Should we mention, that all client certificates (e.g. the ones retrieved via CSRs) need to be renewed as well in this stage?

Comment (rfranzke) [reply docs/usage/shoot_credentials_rotation.md:91]: I got some inspiration from GKE's docs, and there is no explicit mention. But I can do it (as part of #5968 then).

Comment (timebertt) [reply docs/usage/shoot_credentials_rotation.md:91]: > But I can do it (as part of https://github.com/gardener/gardener/pull/5968 then).

Yep, let's add it to the docs to be on the safe side

**Pull request merged**